### ISSUE TITLE
Gentoo ebuild

### ISSUE THREAD

**Issue by hugommg** [StefanYohansson/ktelephone#27]
Hello,

I'm trying to make gentoo ebuilds for ktelephone and pjsip. It's mostly done and ktelephone compiles successfully but when I run it throws the following error:

> alf@wutanglan ~> ktelephone
> 22:27:38.242         os_core_unix.c !pjlib 2.10 for POSIX initialized
> 22:27:40.365         sip_endpoint.c  .Creating endpoint instance...
> 22:27:40.365                  pjlib  .epoll I/O Queue created (0x55e7031f3c30)
> 22:27:40.365         sip_endpoint.c  .Module "mod-msg-print" registered
> 22:27:40.365        sip_transport.c  .Transport manager created.
> 22:27:40.365           pjsua_core.c  .PJSUA state changed: NULL --> CREATED
> 22:27:40.365         sip_endpoint.c  .Module "mod-pjsua-log" registered
> 22:27:40.365         sip_endpoint.c  .Module "mod-tsx-layer" registered
> 22:27:40.365         sip_endpoint.c  .Module "mod-stateful-util" registered
> 22:27:40.365         sip_endpoint.c  .Module "mod-ua" registered
> 22:27:40.365         sip_endpoint.c  .Module "mod-100rel" registered
> 22:27:40.365         sip_endpoint.c  .Module "mod-pjsua" registered
> 22:27:40.366         sip_endpoint.c  .Module "mod-invite" registered
> 22:27:40.904             alsa_dev.c  ..ALSA driver found 9 devices
> 22:27:40.904             alsa_dev.c  ..ALSA initialized
> 22:27:40.904                  pjlib  ..epoll I/O Queue created (0x55e702f5c1f8)
> 22:27:40.905               bcg729.c  ..BCG729 codec initialized
> 22:27:40.907         sip_endpoint.c  .Module "mod-evsub" registered
> 22:27:40.907         sip_endpoint.c  .Module "mod-presence" registered
> 22:27:40.907         sip_endpoint.c  .Module "mod-mwi" registered
> 22:27:40.907         sip_endpoint.c  .Module "mod-refer" registered
> 22:27:40.907         sip_endpoint.c  .Module "mod-pjsua-pres" registered
> 22:27:40.908         sip_endpoint.c  .Module "mod-pjsua-im" registered
> 22:27:40.908         sip_endpoint.c  .Module "mod-pjsua-options" registered
> 22:27:40.908           pjsua_core.c  .1 SIP worker threads created
> 22:27:40.908           pjsua_core.c  .pjsua version 2.10 for Linux-5.6/x86_64/glibc-2.29 initialized
> 22:27:40.908           pjsua_core.c  .PJSUA state changed: CREATED --> INIT
> 22:27:42.089           pjsua_core.c  SIP UDP socket reachable at 192.168.79.130:5090
> 22:27:42.089      udp0x55e702efe200  SIP UDP transport started, published address is 192.168.79.130:5090
> 22:27:42.089           pjsua_core.c  PJSUA state changed: INIT --> STARTING
> 22:27:42.089         sip_endpoint.c  .Module "mod-unsolicited-mwi" registered
> 22:27:42.089           pjsua_core.c  .PJSUA state changed: STARTING --> RUNNING
> *** PJSUA2 STARTED ***
> 22:27:42.089           endpoint.cpp  pjsua_codec_set_priority(&codec_str, priority) error: Not found (PJ_ENOTFOUND) (status=70006) [../src/pjsua2/endpoint.cpp:2198]
> terminate called after throwing an instance of 'pj::Error'
> fish: “ktelephone” terminated by signal SIGABRT (Abort)

Am I missing something?

Thanks

**Comment by StefanYohansson** [comment]
Hey, it is a bug on my side. Need to add a try/catch here

https://github.com/StefanYohansson/ktelephone/blob/571ffa5f0f4d2f28a3eaf8b5d98fd26d51113faa/src/telephony/uamanager.cpp#L18

I will make it until tomorrow, but if you want to give it a try, it should be something like:

```
try {
    ep->codecSetPriority(...);
} catch(Error& err) {
    // i think you dont need to do anything here
}
```

**Comment by StefanYohansson** [comment]
it is a weird error because vanilla pjsua should come with g722 and opus, I guess. Maybe something different on their gentoo build.

**Comment by StefanYohansson** [comment]
Hello, found a similar problem on Ubuntu 20.04, it doesnt provide libpjproject on apt and after compile statically, I hit the same problem. It is now fixed by https://github.com/StefanYohansson/ktelephone/commit/83d65626a3f4ce51424fe24648b2146d29a05a04

**Comment by StefanYohansson** [comment]
@hugommg if you succe

**Comment by hugommg** [comment]
Sorry, I haven't had the time to work on this, it's most likely a missing use flag on some of the dependent libraries. I'll get to it as soon as possible.

**Comment by StefanYohansson** [comment]
yeah, but it is not mandatory, so the try/catch that I add make the ktelephone application works with your current state.
